src/utils/transaction.ts

```diff
@@ -38,6 +42,168 @@ import { CROSS_CHAIN_ACTION_STATUS } from '../constants/transactionDispatcherCon
+        createTimestamp,
+        status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+      }));
+    }
+
+    if (
+      ethers.utils.isAddress(bestRoute.fromToken.address) &&
+      !addressesEqual(bestRoute.fromToken.address, nativeAssetPerChainId[fromChainId].address) &&
+      transactions.length === 1 &&
+      bestRoute.fromAmount
+    ) {
+      const abi = getContractAbi(ContractNames.ERC20Token);
+      const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, bestRoute.fromToken.address);
+      const approvalTransactionRequest = erc20Contract?.encodeApprove?.(transactions[0].to, bestRoute.fromAmount);
+      if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+        return { errorMessage: 'Failed build bridge approval transaction!' };
+      }
+
+      const approvalTransaction = {
+        to: approvalTransactionRequest.to,
+        data: approvalTransactionRequest.data,
+        value: 0,
+        createTimestamp,
+        status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+      };
+
+      transactions = [approvalTransaction, ...transactions];
+    }
+    return { bestRoute, transactions, destinationTxns };
+  } catch (err) {
+    return { errorMessage: 'Failed to get bridge route' };
+  }
+};
+
+const fetchSwapAssetTransaction = async (
+  chainId: number,
+  fromAmountBN: BigNumber,
+  fromAssetAddress: string,
+  offer: ExchangeOffer | null,
+  receiverAddress?: string,
+  sdk?: EtherspotSdk | null
+): Promise<{ errorMessage?: string; result?: { transactions: ICrossChainActionTransaction[] } }> => {
+  if (!sdk) return { errorMessage: 'No sdk found' };
+
+  if (!offer) {
+    return { errorMessage: 'Failed build PLR Dao Staking transaction!' };
+  }
+  try {
+    const createTimestamp = +new Date();
+    let transactions: ICrossChainActionTransaction[] = offer.transactions.map((transaction) => ({
+      ...transaction,
+      chainId,
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    }));
+
+    // not native asset and no erc20 approval transaction included
+    if (
+      fromAssetAddress &&
+      !addressesEqual(fromAssetAddress, nativeAssetPerChainId[chainId].address) &&
+      transactions.length === 1
+    ) {
+      const abi = getContractAbi(ContractNames.ERC20Token);
+      const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, fromAssetAddress);
+      const approvalTransactionRequest = erc20Contract?.encodeApprove?.(transactions[0].to, fromAmountBN);
+      if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+        return { errorMessage: 'Failed build bridge approval transaction!' };
+      }
+
+      const approvalTransaction = {
+        to: approvalTransactionRequest.to,
+        data: approvalTransactionRequest.data,
+        chainId,
+        value: 0,
+        createTimestamp,
+        status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+      };
+
+      transactions = [approvalTransaction, ...transactions];
```

Comment:
For example instead of wrapping the entire function in a try block, I'd break it down into smaller portions like so
```suggestion
let approvalTransactionRequest: [approvalTransactionRequestType];
try {
      const abi = getContractAbi(ContractNames.ERC20Token);
      const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, fromAssetAddress);
      approvalTransactionRequest = erc20Contract?.encodeApprove?.(transactions[0].to, fromAmountBN);
} catch (err) {
      return { errorMessage: 'Failed build transfer transaction!' };
}
      if (!approvalTransactionRequest || !approvalTransactionRequest.to) return { errorMessage: 'Failed build bridge approval transaction!' };

      const approvalTransaction = {
        to: approvalTransactionRequest.to,
        data: approvalTransactionRequest.data,
        chainId,
        value: 0,
        createTimestamp,
        status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
      };

      transactions = [approvalTransaction, ...transactions];
```